src/pages/index.js

```diff
@@ -1,128 +1,213 @@
+  avatarEditForm,
+} from "../constants/constants";
+import Section from "../components/Section";
+
+//const for functions
+let userId;
+let newCardSection;
 
 const api = new Api({
   baseUrl: apiUrl,
   baseHeader: {
     authorization: apiToken,
-    "Content-Type": "application/json"
-  }
+    "Content-Type": "application/json",
+  },
 });
 
-Promise.all([api.getUserInfo()])
-  .then(([userData]) => {
+const cardSelector = selectors.cardTemplate;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cardData]) => {
     userInfo.setUserInfo(userData);
-})
-  //.then(res => res.json())
-  //.then((result) => {
-    //console.log(result);
- // })
+    userId = userData._id;
+    newCardSection = new Section(
+      {
+        items: cardData,
+        renderer: (data) => {
+          const newCard = renderCard(data);
+          newCardSection.addItem(newCard);
+        },
+      },
+      selectors.cardSection
+    );
+    newCardSection.renderItems();
+  })
   .catch(console.error);
 
-const section = new Section({
-  items: initializeCards,
-  renderer: (cardData) => {
-    const cardElement = renderCard(cardData);
-    section.addItem(cardElement);
-    },
-  },
-  cardsList,
-);
-
-section.renderItems();
-
 //edit form
 const userInfo = new UserInfo({
   profileDescriptionSelector: ".profile__description",
   profileNameSelector: ".profile__title",
   profileAvatarSelector: ".profile__img",
-})
+});
 
+/*api
+  .clearAllCards()
+  .then(() => {
+    console.log("All cards have been deleted.");
+  })
+  .catch((error) => {
+    console.error("Failed to delete all cards:", error);
+  });*/
 
-const editFormValidator = new FormValidator(
-  configValidation,
-  profileEditForm
-);
+const editFormValidator = new FormValidator(configValidation, profileEditForm);
 
-const addFormValidator = new FormValidator(
-  configValidation,
-  cardAddForm
-);
+const addFormValidator = new FormValidator(configValidation, cardAddForm);
 
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit
 );
 
-const
-  addNewCardPopup = new PopupWithForm(
-  "#element-add-modal"
-  , handleNewCardSubmit
+const addNewCardPopup = new PopupWithForm(
+  "#element-add-modal",
+  handleNewCardSubmit
 );
 
 const cardPreviewPopup = new PopupWithImage("#preview-modal");
 
+/*const avatarEditPopup = new PopupWithForm(
+  selectors.avatarPopupSelector,
+  handleAvatarFormSubmit
+);*/
 
+function handleAvatarFormSubmit(inputValues) {
+  avatarEditPopup.renderLoading(true);
+  api
+    .updateProfileAvatar(inputValues.avatar)
+    .then(() => {
+      userInfo.setAvatar(inputValues.avatar);
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarEditPopup.renderLoading(false, "Save");
+    });
+}
 
 function handleProfileFormSubmit(inputValues) {
   //profileEditPopup.renderLoading(api);
+  console.log("attempting to submit");
```

Comment:
Needs correction: I don't see the loading on the profile form submit button while the request is in progress